### DISCUSSION TITLE
Pixel 6 Pro doesn't seem to allow connecting more than 10 devices on hotspot? even though it says max 32

### DISCUSSION THREAD

**Discussion by sagz** [Mygod/VPNHotspot#329] [Q&A]
I created a hotspot on a rooted Pixel6Pro (without any VPN, to simplify) and found that after 10 devices are connected, it doesn't allow more client devices to connect. It declines them actively in the client devices showing errors. More so, some existing connected clients also show erratic behavior with unreliable internet connectivity.

VPN Hotspot shows the max as 32:
![Screenshot_20220130-151719](https://user-images.githubusercontent.com/964843/151749853-9666eb46-6f56-4aa4-82ac-c33ffbaf9164.png)

This is what my devices list looks like:
![Screenshot_20220130-151732s](https://user-images.githubusercontent.com/964843/151749903-e435992f-b8de-4904-90b9-c8aa3ff021c7.png)

And settings tab:
![Screenshot_20220130-152320](https://user-images.githubusercontent.com/964843/151749926-4f7d7b6a-d060-4b78-83b3-50205b1f4cf7.png)

I also have the debug log, but it has PII (IP, MACs, VPN app names, etc) so wondering how I can sanitize it or share privately. It's about 20k (!) lines because I've been using the app for a couple months now.

**Comment by sagz** [comment]
Ok, this might not be a VPN Hotspot issue... https://android.stackexchange.com/questions/186487/increase-maximum-number-of-connection-of-wi-fi-hotspot 

Looks like an Android-wide conventional limit of 10.
Why does that denominator say 32 then?

**Reply by Mygod** [reply]
The denominator is reported by your system, in particular the Android Wi-Fi driver. It might not be accurate if there are multiple limits in your system (for example the native Wi-Fi driver, or other parts of the Android Wi-Fi driver).

**Comment by Mygod** [comment]
Uhh. Maybe you should consider getting a proper router. It is cheaper and probably performs better too. :)

**Reply by sagz** [reply]
Yeah I might have to get a router, and that does work pretty well, but I'd like to see if this limit can be removed with root. Any ideas?

**Reply by Mygod** [reply]
As I mentioned in the other thread, there might be multiple places where this limit is put in place. If you figure out what works for you, feel free to share it here.
